torchaudio/prototype/transforms/_transforms.py

```diff
@@ -384,3 +385,56 @@ def forward(self, waveform: torch.Tensor) -> torch.Tensor:
         specgram = self.spectrogram(waveform)
         bark_specgram = self.bark_scale(specgram)
         return bark_specgram
+
+
+def _source_target_sample_rate(orig_freq: int, speed: float) -> Tuple[int, int]:
+    source_sample_rate = int(speed * orig_freq)
+    target_sample_rate = int(orig_freq)
+    gcd = math.gcd(source_sample_rate, target_sample_rate)
+    return source_sample_rate // gcd, target_sample_rate // gcd
+
+
+class TimeStretch(torch.nn.Module):
```

Comment:
what's your take on the name collision with the existing one?

https://pytorch.org/audio/stable/generated/torchaudio.transforms.TimeStretch.html#torchaudio.transforms.TimeStretch

Reply:
ah that's unfortunate. i feel like "time stretch" fits better with the operators introduced here (stretching waveforms in time domain vs stretching spectrograms along time axis). this is also more consistent with librosa (https://librosa.org/doc/main/generated/librosa.effects.time_stretch.html)

Reply:
actually, never mind — librosa's is neither: it accepts a waveform as input, generates spectrograms, and does the stretching on the spectrograms before converting back to time domain

Reply:
let's go with "speed" — consistent with sox and "speed perturbation"